sky/cli.py

```diff
@@ -2692,6 +2692,9 @@ def _down_or_stop_clusters(
         total=len(clusters))
 
     def _down_or_stop(name: str):
+        # if there's a sync process launched by C_SYNC mode running
+        # in the cluster, wait until the sync completes
+        backend_utils.wait_and_terminate_csync(name)
```

Comment:
This adds extra overheads for all the clusters, no matter if it has csync enabled or not. We should probably only apply this for the ones with csync enabled by noting done if the csync is enabled in the local db.

Reply:
Also, should not this be called in the `backend.teardown` or `core.down` instead? For example, `core.down` will not go through this codepath.

Reply:
`wait_and_terminate_csync` now is ran conditionally when the cluster has a storage with `CSYNC` mode according to `state.db`. Also, moved it to `backend.teardown`